src/network/networkd-dhcp4.c

```diff
@@ -583,6 +586,58 @@ static int dhcp4_set_hostname(Link *link) {
         return sd_dhcp_client_set_hostname(link->dhcp_client, hn);
 }
 
+static bool check_promote_secondaries(const char *ifname) {
```

Comment:
`check_<something>` is a bad name: it's not clear that it is a predicate and what the return value is. I think `promote_secondaries_enabled` would be much clearer.